src/execution/as_abi.rs

```diff
@@ -890,20 +759,38 @@ pub(crate) fn assembly_script_local_call(
     }
 }
 
+/// Check whether or not the caller has write access in the current context
+pub fn assembly_caller_has_write_access(env: &ASEnv) -> ABIResult<i32> {
+    sub_remaining_gas(env, settings::metering_local_call_const())?;
+    Ok(env.get_interface().caller_has_write_access()? as i32)
+}
+
+/// Check whether or not the given function exists at the given address
+pub fn assembly_function_exists(env: &ASEnv, address: i32, function: i32) -> ABIResult<i32> {
+    sub_remaining_gas(env, settings::metering_local_call_const())?;
+    let memory = get_memory!(env);
+
+    let address = &get_string(memory, address)?;
+    let function = &get_string(memory, function)?;
+    // NOTE: how do we handle metering for those cases?
+    sub_remaining_gas(env, settings::metering_get_bytecode_const())?;
```

Comment:
This will be removed with gas_calibration dw

Reply:
What is your opinion on the other `NOTE`?

Reply:
I don't know :/

Reply:
Is it a good idea to merge this even though we still have a hard coded value for this? That doesn't seem right